projects/pgai/pgai/vectorizer/embeddings.py

```diff
@@ -404,6 +406,50 @@ async def embed(
                 model_token_length, encoded_documents
             )
 
+    async def create_and_submit_embedding_batch(
+            self,
+            documents: list[dict[str, Any]],
+    ) -> openai.types.Batch:
+        """
+        Creates a batch of embeddings using OpenAI's embeddings API as outlined in
+        https://platform.openai.com/docs/guides/batch/batch-api?lang=python
+
+        Args:
+            documents (list[str]): A list of document chunks to be embedded.
+
+        Returns:
+
+        """
+
+        temp_file = tempfile.NamedTemporaryFile(delete=False, suffix='.jsonl', mode='w')
+
+        for document in documents:
+            entry = {
+                'custom_id': document['unique_full_chunk_id'],
+                'method': 'POST',
+                'url': '/v1/embeddings',
+                'body': {
+                    'model': 'text-embedding-3-small', # TODO how can I use the configured embeddings model?
+                    'input': document['chunk'],
+                },
+            }
+            temp_file.write(json.dumps(entry) + '\n')
+
+        temp_file.close()
+
+        client = openai.OpenAI() # TODO there has to be a client already which I could use instead?
```

Comment:
`self._embedder` is the property that's currently used but I don't think this exposes the batch embedding api.
Probably makes sense to unify this under a simple `self._client` property.

Reply:
So I'd just add that as a new property? AFAIK just initializing will look for credentials from the environment, is that what we want?